library/database/class.database.php

```diff
@@ -150,11 +150,15 @@ public function commitTransaction() {
      */
     protected function newPDO($Dsn, $User, $Password) {
         try {
+            if (strpos($Dsn, 'charset=') === false) {
```

Comment:
I don't think this is the right place for this code. You should be adding the `charset` when the DSN is being constructed.